source/common/upstream/cluster_manager_impl.h

```diff
@@ -232,9 +233,14 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
    */
   struct ThreadLocalClusterManagerImpl : public ThreadLocal::ThreadLocalObject {
     struct ConnPoolsContainer {
-      typedef std::map<std::vector<uint8_t>, Http::ConnectionPool::InstancePtr> ConnPools;
+      ConnPoolsContainer(Event::Dispatcher& dispatcher)
+          : pools_{std::make_shared<ConnPools>(dispatcher)} {}
 
-      ConnPools pools_;
+      typedef ConnPoolMap<std::vector<uint8_t>, Http::ConnectionPool::Instance> ConnPools;
+
+      // this is a shared_ptr so we can keep it alive while cleaning up
```

Comment:
nit: Start comment with capital, end with full stop, etc. Same elsewhere.

Reply:
Yeah, sorry about that. I try to use proper grammar, but fall into bad habits sometimes. :)